### PR TITLE
Use versioning crate from operator-rs

### PR DESCRIPTION
**Pull request by maltesander**
## Description

- Using versioning from operator-rs
- Removed obsolete code
- Adapted tests

## Review Checklist
- [ ] Code contains useful comments
- [ ] (Integration-)Test cases added (or not applicable)
- [ ] Documentation added (or not applicable)
- [ ] Changelog updated (or not applicable)

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -4,14 +4,32 @@ All notable changes to this project will be documented in this file.
...
+- Code for version handling ([#210]).
+- Removed `current_version` and `target_version` from cluster status ([#210]).
+
+[#210]: https://github.com/stackabletech/zookeeper-operator/pull/210
+
 ## [0.2.0] - 2021-09-14
 
+### Added
+- Added versioning code from operator-rs for up and downgrades ([#210]).
+- Added `ProductVersion` to status ([#210]).
```
Comment (razvan) [inline CHANGELOG.md:21]: What happend here ? Surely these changes were not already done in release 0.2.0, right ?

File: CHANGELOG.md [hunk 2]
```diff
@@ -4,14 +4,32 @@ All notable changes to this project will be documented in this file.
...
 
 ### Changed
 - **Breaking:** Repository structure was changed and the -server crate renamed to -binary. As part of this change the -server suffix was removed from both the package name for os packages and the name of the executable ([#197]).
 
-## [0.1.0] - 2021.09.07
+### Removed
+- Code for version handling ([#210]).
+- Removed `current_version` and `target_version` from cluster status ([#210]).
```
Comment (razvan) [inline CHANGELOG.md:28]: This is wrong.

File: rust/operator-binary/Cargo.toml [hunk 3]
```diff
@@ -9,7 +9,7 @@ version = "0.3.0-nightly"
 build = "build.rs"
 
 [dependencies]
-stackable-operator = { git = "https://github.com/stackabletech/operator-rs.git", tag = "0.1.0" }
+stackable-operator = { git = "https://github.com/stackabletech/operator-rs.git", branch = "main" }
```
Comment (razvan) [inline rust/operator-binary/Cargo.toml:12]:
```suggestion
stackable-operator = { git = "https://github.com/stackabletech/operator-rs.git", tag = "0.2.0" }
```

File: rust/operator-binary/Cargo.toml [hunk 4]
```diff
@@ -20,7 +20,7 @@ tracing = "0.1"
 
 [build-dependencies]
 built = { version =  "0.5", features = ["chrono", "git2"] }
-stackable-operator = { git = "https://github.com/stackabletech/operator-rs.git", tag = "0.1.0" }
+stackable-operator = { git = "https://github.com/stackabletech/operator-rs.git", branch = "main" }
```
Comment (razvan) [inline rust/operator-binary/Cargo.toml:23]:
```suggestion
stackable-operator = { git = "https://github.com/stackabletech/operator-rs.git", tag = "0.2.0" }
```

File: rust/operator/Cargo.toml [hunk 5]
```diff
@@ -9,7 +9,7 @@ version = "0.3.0-nightly"
 
 [dependencies]
 product-config = { git = "https://github.com/stackabletech/product-config.git", tag = "0.1.0" }
-stackable-operator = { git = "https://github.com/stackabletech/operator-rs.git", tag = "0.1.0" }
+stackable-operator = { git = "https://github.com/stackabletech/operator-rs.git", branch = "main" }
```
Comment (razvan) [inline rust/operator/Cargo.toml:12]:
```suggestion
stackable-operator = { git = "https://github.com/stackabletech/operator-rs.git", tag = "0.2.0" }
```

**Review by razvan — COMMENTED** [review]
The change log doesn't look right.

File: CHANGELOG.md (hunk 1 above)
Comment (maltesander) [reply CHANGELOG.md]: yes copy pasted it wrong and did not remove it...

**Review by razvan — APPROVED** [review]
Updating the condition works. The integration tests fail but that doesn't seem to be related to this change.

**Pull request merged**